shared_model/interfaces/iroha_internal/block.hpp

```diff
@@ -54,53 +52,58 @@ namespace shared_model {
        */
       virtual HashType &merkleRoot() const = 0;
 
+      /// Type of a single Transaction
+      using TransactionType = detail::PolymorphicWrapper<Transaction>;
+
       /// Type of transactions' collection
-      using TransactionsType = std::vector<Transaction>;
+      using TransactionsCollectionType = std::vector<TransactionType>;
 
       /**
        * @return collection of transactions
        */
-      virtual TransactionsType &transactions() const = 0;
+      virtual const TransactionsCollectionType &transactions() const = 0;
 
       iroha::model::Block *makeOldModel() const override {
         iroha::model::Block *oldStyleBlock = new iroha::model::Block();
         oldStyleBlock->height = height();
         oldStyleBlock->prev_hash =
-            iroha::blob_t<256 / 8>::from_string(prevHash().toString());
+            iroha::model::Block::HashType::from_string(prevHash().toString());
         oldStyleBlock->txs_number = txsNumber();
         oldStyleBlock->merkle_root =
-            iroha::blob_t<256 / 8>::from_string(merkleRoot().toString());
-        for (auto &tx : transactions()) {
-          oldStyleBlock->transactions.push_back(*tx->makeOldModel());
-        }
+            iroha::model::Block::HashType::from_string(merkleRoot().toString());
+        std::for_each(
+            transactions().begin(),
+            transactions().end(),
+            [oldStyleBlock](auto &tx) {
```

Comment:
Does it work? Seems no for me. Capture by ref, and it would be nice to write tests to ensure everything goes right

Reply:
```oldStyleBlock``` is a pointer so it should, I think.